anta/tests/interfaces.py

```diff
@@ -1055,3 +1055,109 @@ def test(self) -> None:
+          traffic_classes:
+            - TC0
+            - TC3
+          ignored_interfaces:
+            - Ethernet
+            - Port-Channel1
+    ```
+    """
+
+    categories: ClassVar[list[str]] = ["interfaces"]
+    commands: ClassVar[list[AntaCommand | AntaTemplate]] = [AntaCommand(command="show interfaces counters queue detail", revision=1)]
+
+    class Input(AntaTest.Input):
+        """Input model for the VerifyInterfaceQueuDrops test."""
+
+        traffic_classes: list[str] | None = None
+        """List of traffic classes to be verified. If None, all available traffic classes will be checked."""
+        traffic_type: Literal["unicast", "multicast"] | None = None
+        """Type of network traffic. If None, both will be checked."""
+        interfaces: list[Interface] | None = None
+        """A list of interfaces to be tested. If not provided, all interfaces (excluding any in `ignored_interfaces`) are tested."""
+        ignored_interfaces: list[InterfaceType | Interface] | None = None
+        """A list of interfaces or interface types like Management which will ignore all Management interfaces."""
+        dropped_pckt_threshold: PositiveInteger = 0
+        """Threshold for the number of dropped packets."""
+
+    def _verify_dropped_packets(
+        self,
+        interface: Interface,
+        traffic_type: Literal["unicast", "multicast"] | None,
+        traffic_classes: list[str] | None,
+        output: dict[str, Any],
+        threshold: PositiveInteger,
+    ) -> None:
+        """Verify Egress/Ingress dropped packets(unicast/multicast) for an input interface and traffic class."""
+        unicast_class_details = output.get("ucastQueues", {}).get("trafficClasses", {})
+        unicast_traffic_cls = traffic_classes if traffic_classes else unicast_class_details.keys()
+        multicast_class_details = output.get("mcastQueues", {}).get("trafficClasses", {})
+        multicast_traffic_cls = traffic_classes if traffic_classes else multicast_class_details.keys()
+
+        if not traffic_type:
+            # Verify the unicast traffic type
+            self._verify_traffic_class_details(interface, "Unicast", unicast_traffic_cls, unicast_class_details, threshold)
+            # Verify the multicast traffic type
+            self._verify_traffic_class_details(interface, "Multicast", multicast_traffic_cls, multicast_class_details, threshold)
+
+        # Verify the given taffic type
+        elif traffic_type == "unicast":
+            self._verify_traffic_class_details(interface, traffic_type, unicast_traffic_cls, unicast_class_details, threshold)
+
+        elif traffic_type == "multicast":
+            self._verify_traffic_class_details(interface, traffic_type, multicast_traffic_cls, multicast_class_details, threshold)
+
+    def _verify_traffic_class_details(
+        self, interface: Interface, network_type: str, traffic_classes: list[str], output: dict[str, Any], threshold: PositiveInteger
+    ) -> None:
```

Comment:
Do we need this helper?
```suggestion
    def _verify_traffic_class_details(
        self, interface: Interface, network_type: str, traffic_classes: list[str], output: dict[str, Any]
    ) -> None:
```

Reply:
Updated utils.